Client/Frontend/Browser/UserScriptManager.swift

```diff
@@ -106,6 +115,22 @@ class UserScriptManager {
         return WKUserScript(source: alteredSource, injectionTime: .atDocumentStart, forMainFrameOnly: false)
     }()
     
+    // PaymentRequestUserScript is injected at document start to handle
+    // requests to payment APIs
+    private let PaymentRequestUserScript: WKUserScript? = {
+        guard let path = Bundle.main.path(forResource: "PaymentRequest", ofType: "js"), let source = try? String(contentsOfFile: path) else {
+            log.error("Failed to load PaymentRequest.js")
+            return nil
+        }
+        
+        var alteredSource = source
+        let token = UserScriptManager.securityToken.uuidString.replacingOccurrences(of: "-", with: "", options: .literal)
+        alteredSource = alteredSource.replacingOccurrences(of: "$<paymentreq>", with: "PaymentRequest\(token)", options: .literal)
+        alteredSource = alteredSource.replacingOccurrences(of: "$<paymentreqcallback>", with: "PaymentRequestCallback\(token)", options: .literal)
+        
+        return WKUserScript(source: alteredSource, injectionTime: .atDocumentStart, forMainFrameOnly: false)
```

Comment:
Is there a reason this needs to be injected at the start?

Reply:
`PaymentRequest` can be generated before load.